### ISSUE TITLE
[Bug]  [mmdeploy] [error] [net_module.cpp:40] Net backend not found: torchscript 请问 SDK支持 C++  torchscript 吗？

### ISSUE THREAD

**Issue by 1095788063** [open-mmlab/mmdeploy#1627]
### Checklist

- [X] I have searched related issues but cannot get the expected help.
- [X] 2. I have read the [FAQ documentation](https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/faq.md) but cannot get the expected help.
- [X] 3. The bug has not been fixed in the latest version.

请问 SDK支持 C++  torchscript 吗？运行就提示不支持和报错。我的电脑环境都有安装好的了，模型也转换成功了。

[2023-01-09 23:28:52.573] [mmdeploy] [info] [model.cpp:35] [DirectoryModel] Load model: "D:\ayjdata\Code\Deep_learning\OpenMMLab\mmdetection\mmdetection_V3.0.0rc4_dev-3.x_2022-12-10\runs\train\mask-rcnn_r50_fpn_amp-1x_coco_CL_2022-12-16\torchscript"
[2023-01-09 23:28:52.748] [mmdeploy] [info] [inference.cpp:44] {
  "context": {
    "device": "<any>",
    "model": "<any>",
    "stream": "<any>"
  },
  "pipeline": {
    "input": [
      "img"
    ],
    "output": [
      "post_output"
    ],
    "tasks": [
      {
        "input": [
          "img"
        ],
        "module": "Transform",
        "name": "Preprocess",
        "output": [
          "prep_output"
        ],
        "transforms": [
          {
            "file_client_args": {
              "backend": "disk"
            },
            "type": "LoadImageFromFile"
          },
          {
            "keep_ratio": true,
            "size": [
              640,
              640
            ],
            "type": "Resize"
          },
          {
            "mean": [
              123.675,
              116.28,
              103.53
            ],
            "std": [
              58.395,
              57.12,
              57.375
            ],
            "to_rgb": true,
            "type": "Normalize"
          },
          {
            "size_divisor": 32,
            "type": "Pad"
          },
          {
            "type": "DefaultFormatBundle"
          },
          {
            "keys": [
              "img"
            ],
            "meta_keys": [
              "flip_direction",
              "img_id",
              "scale_factor",
              "img_shape",
              "ori_shape",
              "img_norm_cfg",
              "ori_filename",
              "filename",
              "pad_param",
              "pad_shape",
              "valid_ratio",
              "img_path",
              "flip"
            ],
            "type": "Collect"
          }
        ],
        "type": "Task"
      },
      {
        "input": [
          "prep_output"
        ],
        "input_map": {
          "img": "#0"
        },
        "is_batched": false,
        "module": "Net",
        "name": "maskrcnn",
        "output": [
          "infer_output"
        ],
        "output_map": {
          "dets": "#0",
          "labels": "#1",
          "masks": "#2"
        },
        "type": "Task"
      },
      {
        "component": "ResizeInstanceMask",
        "input": [
          "prep_output",
          "infer_output"
        ],
        "module": "mmdet",
        "name": "postprocess",
        "output": [
          "post_output"
        ],
        "params": {
          "mask_thr_binary": 0.5,
          "min_bbox_size": 0,
          "rcnn": {
            "mask_thr_binary": 0.5,
            "max_per_img": 100,
            "nms": {
              "iou_threshold": 0.5,
              "type": "nms"
            },
            "score_thr": 0.05
          },
          "rpn": {
            "max_per_img": 1000,
            "min_bbox_size": 0,
            "nms": {
              "iou_threshold": 0.7,
              "type": "nms"
            },
            "nms_pre": 1000
          },
          "score_thr": 0.05
        },
        "type": "Task"
      }
    ]
  }
}
[2023-01-09 23:28:52.749] [mmdeploy] [error] [net_module.cpp:40] Net backend not found: torchscript, available backends: [("tensorrt", 0)]
[2023-01-09 23:28:52.893] [mmdeploy] [error] [task.cpp:99] error parsing config: {
  "context": {
    "device": "<any>",
    "model": "<any>",
    "stream": "<any>"
  },
  "input": [
    "prep_output"
  ],
  "input_map": {
    "img": "#0"
  },
  "is_batched": false,
  "module": "Net",
  "name": "maskrcnn",
  "output": [
    "infer_output"
  ],
  "output_map": {
    "dets": "#0",
    "labels": "#1",
    "masks": "#2"
  },
  "type": "Task"
}
[2023-01-09 23:28:52.893] [mmdeploy] [info] [inference.cpp:56] ["img"] <- ["image"]
[2023-01-09 23:28:52.893] [mmdeploy] [info] [inference.cpp:67] ["post_output"] -> ["dets"]

**Comment by 1095788063** [comment]
![image](https://user-images.githubusercontent.com/29217820/211346220-f058fd5b-0fd9-41dc-9703-83d4af55eee6.png)

![image](https://user-images.githubusercontent.com/29217820/211346387-a8bdd3ad-64f4-4f3a-918c-1d8c7568b870.png)

![image](https://user-images.githubusercontent.com/29217820/211346465-a86488e1-d831-4c0c-afac-e58f50f7932f.png)


01/09 23:33:03 - mmengine - INFO - **********Environmental information**********
01/09 23:33:06 - mmengine - INFO - sys.platform: win32
01/09 23:33:06 - mmengine - INFO - Python: 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)]
01/09 23:33:06 - mmengine - INFO - CUDA available: True
01/09 23:33:06 - mmengine - INFO - numpy_random_seed: 2147483648
01/09 23:33:06 - mmengine - INFO - GPU 0: NVIDIA GeForce RTX 3070
01/09 23:33:06 - mmengine - INFO - CUDA_HOME: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.3
01/09 23:33:06 - mmengine - INFO - NVCC: Cuda compilation tools, release 11.3, V11.3.58
01/09 23:33:06 - mmengine - INFO - MSVC: 用于 x64 的 Microsoft (R) C/C++ 优化编译器 19.29.30147 版
01/09 23:33:06 - mmengine - INFO - GCC: n/a
01/09 23:33:06 - mmengine - INFO - PyTorch: 1.10.0+cu113
01/09 23:33:06 - mmengine - INFO - PyTorch compiling details: PyTorch built with:
  - C++ Version: 199711
  - MSVC 192829337
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 2019
  - LAPACK is enabled (usually provided by MKL)
  - CPU capability usage: AVX2
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.2
  - Magma 2.5.4
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.2.0, CXX_COMPILER=C:/w/b/windows/tmp_bin/sccache-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -openmp:experimental -IC:/w/b/windows/mkl/include -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, 

01/09 23:33:06 - mmengine - INFO - TorchVision: 0.11.0+cu113
01/09 23:33:06 - mmengine - INFO - OpenCV: 4.5.5
01/09 23:33:06 - mmengine - INFO - MMEngine: 0.4.0
01/09 23:33:06 - mmengine - INFO - MMCV: 2.0.0rc3
01/09 23:33:06 - mmengine - INFO - MMCV Compiler: MSVC 192829924
01/09 23:33:06 - mmengine - INFO - MMCV CUDA Compiler: 11.3
01/09 23:33:06 - mmengine - INFO - MMDeploy: 1.0.0rc1+
01/09 23:33:06 - mmengine - INFO - 

01/09 23:33:06 - mmengine - INFO - **********Backend information**********
01/09 23:33:06 - mmengine - INFO - tensorrt:	8.2.5.1
01/09 23:33:06 - mmengine - INFO - tensorrt custom ops:	Available
01/09 23:33:06 - mmengine - INFO - ONNXRuntime:	1.8.1
01/09 23:33:06 - mmengine - INFO - ONNXRuntime-gpu:	None
01/09 23:33:06 - mmengine - INFO - ONNXRuntime custom ops:	NotAvailable
01/09 23:33:06 - mmengine - INFO - pplnn:	None
01/09 23:33:06 - mmengine - INFO - ncnn:	None
01/09 23:33:06 - mmengine - INFO - snpe:	None
01/09 23:33:06 - mmengine - INFO - openvino:	None
01/09 23:33:06 - mmengine - INFO - torchscript:	1.10.0+cu113
01/09 23:33:06 - mmengine - INFO - torchscript custom ops:	Available
01/09 23:33:07 - mmengine - INFO - rknn-toolkit:	None
01/09 23:33:07 - mmengine - INFO - rknn2-toolkit:	None
01/09 23:33:07 - mmengine - INFO - ascend:	None
01/09 23:33:07 - mmengine - INFO - coreml:	None
01/09 23:33:07 - mmengine - INFO - tvm:	None
01/09 23:33:07 - mmengine - INFO - 

01/09 23:33:07 - mmengine - INFO - **********Codebase information**********
01/09 23:33:07 - mmengine - INFO - mmdet:	3.0.0rc5
01/09 23:33:07 - mmengine - INFO - mmseg:	None
01/09 23:33:07 - mmengine - INFO - mmcls:	None
01/09 23:33:07 - mmengine - INFO - mmocr:	None
01/09 23:33:07 - mmengine - INFO - mmedit:	None
01/09 23:33:07 - mmengine - INFO - mmdet3d:	None
01/09 23:33:07 - mmengine - INFO - mmpose:	None
01/09 23:33:07 - mmengine - INFO - mmrotate:	None
01/09 23:33:07 - mmengine - INFO - mmaction:	None

进程已结束,退出代码0

**Comment by 1095788063** [comment]
object_detection.cpp 
device_name=cuda 
model_path="torchscript"
image_path=test.jpg

**Comment by irexyc** [comment]
When have not tested on windows. To have a try, you need to build torchvision from source to support some operations like nms. When build sdk, you have to seg `MMDEPLOY_TORCHSCRIPT_SDK_BACKEND` to `ON` and add `torchscript` to `MMDEPLOY_TARGET_BACKENDS`.

**Comment by 1095788063** [comment]
> When have not tested on windows. To have a try, you need to build torchvision from source to support some operations like nms. When build sdk, you have to seg `MMDEPLOY_TORCHSCRIPT_SDK_BACKEND` to `ON` and add `torchscript` to `MMDEPLOY_TARGET_BACKENDS`.



> When have not tested on windows. To have a try, you need to build torchvision from source to support some operations like nms. When build sdk, you have to seg `MMDEPLOY_TORCHSCRIPT_SDK_BACKEND` to `ON` and add `torchscript` to `MMDEPLOY_TARGET_BACKENDS`.

https://github.com/open-mmlab/mmdeploy/blob/master/docs/zh_cn/01-how-to-build/cmake_option.md 这里没有写更新说明，所以不知道这个参数配置。

**Comment by 1095788063** [comment]
> When have not tested on windows. To have a try, you need to build torchvision from source to support some operations like nms. When build sdk, you have to seg `MMDEPLOY_TORCHSCRIPT_SDK_BACKEND` to `ON` and add `torchscript` to `MMDEPLOY_TARGET_BACKENDS`.

python转换成功了，但是C++ 的 SDK还是不行。
![image](https://user-images.githubusercontent.com/29217820/211522244-ef6500f4-7ee2-4e81-b792-86e99efb1c8f.png)

![image](https://user-images.githubusercontent.com/29217820/211522334-09f8d66d-08a8-4718-8c12-10ec90be0bd9.png)

[2023-01-10 18:06:07.368] [mmdeploy] [error] [net_module.cpp:40] Net backend not found: torchscript, available backends: [("tensorrt", 0)]
[2023-01-10 18:06:07.470] [mmdeploy] [error] [task.cpp:99] error parsing config: {
  "context": {
    "device": "<any>",
    "model": "<any>",
    "stream": "<any>"
  },
  "input": [
    "prep_output"
  ],
  "input_map": {
    "img": "#0"
  },
  "is_batched": false,
  "module": "Net",
  "name": "maskrcnn",
  "output": [
    "infer_output"
  ],
  "output_map": {
    "dets": "#0",
    "labels": "#1",
    "masks": "#2"
  },
  "type": "Task"
}
[2023-01-10 18:06:07.471] [mmdeploy] [info] [inference.cpp:56] ["img"] <- ["image"]
[2023-01-10 18:06:07.471] [mmdeploy] [info] [inference.cpp:67] ["post_output"] -> ["dets"]

**Comment by irexyc** [comment]
Closed as the `Net backend not found: torchscript` error seems solved according to https://github.com/open-mmlab/mmdeploy/issues/1642
